climada/entity/exposures/crop_production.py

```diff
@@ -142,19 +155,38 @@ def set_from_single_run(self, input_dir=INPUT_DIR, filename=None, hist_mean=HIST
                 f.i. 'mai', 'ric', 'whe', 'soy'
             irr (string): irrigation type
                 f.i 'firr' (full irrigation), 'noirr' (no irrigation) or 'combined'= firr+noirr
+            isimip_version(str): 'ISIMIP2' (default) or 'ISIMIP3'
             unit (string): unit of the exposure (per year)
-                f.i 'USD' or 't'
+                f.i 'USD' or 't' (default) or 'kcal'
             fn_str_var (string): FileName STRing depending on VARiable and
                 ISIMIP simuation round
 
         Returns:
             Exposure
         """
-
+        if not bbox:
+            bbox = BBOX
+        if not input_dir:
+            input_dir = INPUT_DIR
+        if hist_mean is None:
+            hist_mean = HIST_MEAN_PATH
+        if not fn_str_var:
+            fn_str_var = FN_STR_VAR
+        if (not isimip_version) or ('ISIMIP2' in isimip_version):
```

Comment:
That is strange. Why do you redefine `isimip_version` when `'ISIMIP2` is in `isimip_version?

Reply:
This is to allow for inputs for isimip_version like 'ISIMIP2a' instead of 'ISIMIP2'. If input is 'ISIMIP2a', it is automatically reduced to 'ISIMIP2'. I would like to keep that tolerance in for convenience of users, because in the module relative_cropyield, the differenciation between **a** and **b** plays a role, while in crop_production it doesn't....

How about if I change it to the following to be less confusing, @chahank?

`if (not isimip_version) or (isimip_version in ['ISIMIP2a, 'ISIMIP2b']):`